components/ide/jetbrains/backend-plugin/build.gradle-stable.kts

```diff
@@ -34,15 +34,30 @@ project(":") {
     kotlin {
         val excludedPackage = if (environmentName == "latest") "stable" else "latest"
         sourceSets["main"].kotlin.exclude("io/gitpod/jetbrains/remote/${excludedPackage}/**")
+
+        if (properties("platformType") == "RD") {
+            print("Rider: exclude unnecessary files")
+            sourceSets["main"].kotlin.exclude("**/GitpodForceUpdateMavenProjectsActivity.kt")
+            sourceSets["main"].kotlin.exclude("**/maven.xml")
+        }
     }
 
     sourceSets {
         main {
-            resources.srcDirs("src/main/resources-${environmentName}")
+            resources.srcDirs("src/main/resources")
+            if (properties("platformType") == "RD") {
+                print("Rider: import rider source set")
+                resources.srcDirs("src/main/resources-rider")
+            }
+            resources.srcDirs("src/main/resources-${environmentName.replace("-rider", "")}")
         }
     }
 }
 
+tasks.named<ProcessResources>("processResources") {
+    duplicatesStrategy = DuplicatesStrategy.WARN
```

Comment:
It seems the default value is INHERIT, and if there's no parent it defaults to INCLUDE ([ref](https://github.com/gradle/gradle/blob/8ca9ccc0e067fa643bb1f55c239e7b4d85a02500/subprojects/core-api/src/main/java/org/gradle/api/file/DuplicatesStrategy.java#L24-L30)).

By setting warn, it means we'll take the most recent file.

Is it okay to take the most recent file? What files are we going to skip/ignore by taking the latest?

Reply:
You can try to comment these lines, build complains logo.svg duplicated, which with `sourceSet` settings, this should no happen. This change is about to fix it